### PR TITLE
[Owner Review] Added more missing enums to NI-DMM

### PR DESCRIPTION
**Pull request by ShantanuShinde**
### What does this Pull Request accomplish?

Added metadata of more of the missing enums from the `nidmm.h` and generated them. Also attribute `INSTRUMENT_PRODUCT_ID` renamed to `NIDMM_ATTRIBUTE_CONFIG_PRODUCT_NUMBER` to match documentation

### Why should this Pull Request be merged?

The enums are used in attributes and APIs as an alternative to passing values directly

### What testing has been done?

The enums built and generated successfully and the read-write ones have been tested with API system tests

### CODE REVIEW

File: generated/nidmm/nidmm.proto [hunk 1]
```diff
@@ -346,6 +362,13 @@ enum MeasurementCompleteDest {
   MEASUREMENT_COMPLETE_DEST_NIDMM_VAL_LBR_TRIG0 = 1003;
 }
 
+enum MeasurementDestinationSlope {
```
Comment (Akshata440) [inline generated/nidmm/nidmm.proto:387]: You could combine SampleTrigSlope and TriggerSlope with this and have just one enum that contains VAL_POSITIVE and VAL_NEGATIVE

**Review by Akshata440 — CHANGES_REQUESTED** [review]

File: generated/nidmm/nidmm.proto (hunk 1 above)
Comment (ShantanuShinde) [reply generated/nidmm/nidmm.proto:387]: combined them

**Review by Akshata440 — APPROVED** [review]

File: source/codegen/metadata/nidmm/enums.py [hunk 2]
```diff
@@ -544,7 +564,7 @@
     'ThermocoupleReferenceJunctionType': {
         'values': [
             {
-                'name': 'NIDMM_VAL_FIXED',
+                'name': 'NIDMM_VAL_TEMP_REF_JUNC_FIXED',
                 'value': 2
```
Comment (doshirohan) [inline source/codegen/metadata/nidmm/enums.py:652]: Looks like this name change is not captured in CHANGES.md

**Review by doshirohan — APPROVED** [review]

File: source/codegen/metadata/nidmm/functions.py [hunk 3]
```diff
@@ -696,7 +696,8 @@
         {
             "name": "sampleTriggerSlope",
             "direction": "in",
-            "type": "ViInt32"
+            "type": "ViInt32",
+            "enum": "Slopes"
```
Comment (jonlugoatx) [inline source/codegen/metadata/nidmm/functions.py]: Question: is this information encoded in the base hapigen metadata somewhere?

**Review by reckenro — APPROVED** [review]

File: source/codegen/metadata/nidmm/functions.py (hunk 3 above)
Comment (ShantanuShinde) [reply source/codegen/metadata/nidmm/functions.py]: that one 3 of the enums that are in hapigen were combined as they have same values, but now that change is reverted to keep consistency with the enums.hapigen

File: source/codegen/metadata/nidmm/enums.py (hunk 2 above)
Comment (ShantanuShinde) [reply source/codegen/metadata/nidmm/enums.py:652]: updated changes.md with this change

**Review by jonlugoatx — APPROVED** [review]

**Pull request merged**